### DISCUSSION TITLE
mqtt linkquality value question

### DISCUSSION THREAD

**Discussion by sammmuel** [Koenkk/zigbee2mqtt#5684] [Devices]
Hello,
I have Xiaomi aquara temperature sensors, a CC2531 adapter on my raspberry and I recently added a CC2530 router to improve the quality of the network.
It works well, I don't have any more disconnection problems on some sensors as it used to happen before.
On the other hand, I saw with MQTT-explorer that my router and all my sensors that pass through this router have the same linkquality value.
So I'm a bit lost, I think that only the last link between the router and my raspberry is evaluated and not the whole link.
To summarize here is my problem, until now I used the linkquality value to detect the loss of a sensor (if linkquality=0) and I cut the regulation of my heating at this place accordingly. Now I believe that this may not work anymore.
If someone has an alternative solution I'll take it.
Thank you in advance.

**Comment by Koenkk** [comment]
> So I'm a bit lost, I think that only the last link between the router and my raspberry is evaluated and not the whole link.

That's indeed true, the Zigbee messages do not contain the linkqualty from end device -> router so we cannot send it.

**Comment by sigma415** [comment]
@Koenkk : Any chance that Z2M (some day) can handle linkquality value from end devices over routers ?

**Reply by Koenkk** [reply]
No, because the linkquality is not included in the message received by the coordinator. You will need to use the network map functionality for that.

**Reply by psiberfunk** [reply]
How does the network map handle that any better ?  I see the same problem there with a centralite router and my motion senors all pegged to 255.. is that a deficiency of this particular router or another zigbee protocol limitation ?  

Example map included:

<img width="973" height="410" alt="image" src="https://github.com/user-attachments/assets/8d8e2beb-9a57-4116-9073-6e97bf0a4360" />
